test/e2e/tests/setup_e2e_test.go

```diff
@@ -147,6 +116,40 @@ var _ = AfterSuite(func() {
 	}
 })
 
+func NewTestContext(zone string) *remote.TestContext {
```

Comment:
Reviewer note: I was experimenting with spinning up a new instance for my test case without it being added to the global pool of test contexts. I didn't end up using it, but the refactor allows future test cases to perform a similar setup.